src/pages/api/logs/search.js

```diff
@@ -9,6 +10,14 @@ const logParams = z.object({
   dog: z.string().refine((id) => {
     return Types.ObjectId.isValid(id) ? new Types.ObjectId(id) : null;
   }),
+  query: z.string().optional(),
+  filters: z
+    .object({
+      topic: z.enum(consts.topicArray).optional(),
+      severity: z.enum(consts.concernArray).optional(),
```

Comment:
topic and severity should be arrays of enums

Reply:
My bad, my last change to the Zod schema broke the filtering and I apparently forgot to test it 😐 

My latest commit should fix it. I realized that making topic and severity arrays of enums wasn't enough because the frontend apparently wasn't sending arrays at all, it was sending objects whose keys were integers that kind of acted like arrays - not sure why this is but it's something to do with how the appliedFilters variable is set/modified. So I also added some code on the frontend to convert its members to arrays before making the post request. Hope that's not a problem